db/src/lib.rs

```diff
@@ -35,6 +32,34 @@ pub async fn create_user(pool: &PgPool, user_name: &str) -> color_eyre::Result<U
     Ok(result.user_id)
 }
 
+pub async fn add_url(pool: &PgPool, url: &str, allow_existing: &bool) -> color_eyre::Result<String> {
+    let exist_record = sqlx::query_as!(
+        ExistRecord,
+        "SELECT EXISTS(SELECT 1 FROM Page WHERE url = $1) as exists",
+        url
+    )
+    .fetch_one(pool)
+    .await?;
```

Comment:
exist_record would return true or false. The reason why is that the system checks if the user passed "-- allow existing" with either true or false

Reply:
Totally makes sense!

In cases like this where you need to do a SQL query, and just need the result within your function you might find it easier to use `query!` as opposed to `query_as!`

When you use `query!` you get a 'Record' back that magically has the columns and types that the SQL returned!
The downside is that its not a 'named struct' so you can't really pass it around nicely. But for something like this where you just need to use the boolean I think it might work nicely!